### PR TITLE
Jay s3 1

### PR DESCRIPTION
**Pull request by jayliu223**
This PR consists of the following changes:

- inference.py has been simplified and modularized.
- the ML model has been integrated into main.py and initializes along with Flask app.
- a basic setup has been created where probability is displayed next to text input (shown below).

image

### CODE REVIEW

File: inference.py [hunk 1]
```diff
@@ -1,36 +1,31 @@
 import h5py
 import json
-import numpy as np
-import pandas as pd
-from sklearn.model_selection import train_test_split
 from keras_preprocessing.text import Tokenizer, tokenizer_from_json
```
Comment (MohammadOsama15) [inline inference.py]: Can you change this line to : "from keras.preprocessing.text import Tokenizer, tokenizer_from_json"

File: inference.py [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
 from keras_preprocessing.sequence import pad_sequences
```
Comment (MohammadOsama15) [inline inference.py]: Can you change this line to : from keras.utils import pad_sequences

File: main.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 from flask_login import login_required, current_user
 from . import db
 from .redditAPI import search_posts
+from inference import model, tokenize_sequence
```
Comment (MohammadOsama15) [inline main.py]: can you change this please to "from .inference import model, tokenize_sequence"

**Review by MohammadOsama15 — CHANGES_REQUESTED** [review]

File: main.py (hunk 3 above)
Comment (jayliu223) [reply main.py]: Hey Mohammad,

I have implemented the changes you've requested.

**Review by MohammadOsama15 — APPROVED** [review]
Looks good! Thank you for your change!

**Pull request merged**